src/utils/application.data.ts

```diff
@@ -42,3 +43,28 @@ export const mapFieldsToDataObject = (data: any, htmlFields: string[], dataModel
 export const mapDataObjectToFields = (data: any, htmlFields: string[], dataModelKeys: string[]) => {
   return htmlFields.reduce((o, key, i) => Object.assign(o, { [key]: data[dataModelKeys[i]] }), {});
 };
+
+export const removeFromApplicationData = (session: Session | undefined, key: string, id: string | undefined) => {
+  const appData: ApplicationData = getApplicationData(session);
+
+  if (id && appData && appData[key]) {
+    const index = appData[key].findIndex( object => object[ID] === id );
+
+    if (index !== -1) {
+      appData[key].splice(index, 1);
+      setExtraData(session, appData);
+    }
+  }
+};
+
+export const getFromApplicationData = (session: Session | undefined, key: string, id: string | undefined): any => {
+  const appData: ApplicationData = getApplicationData(session);
+
+  if (id && appData && appData[key]) {
+    const index = appData[key].findIndex(object => object[ID] === id );
+
+    return appData[key][index] || {};
+  }
+
+  return {};
```

Comment:
What are the consequences of returning this empty structure here? If every time this method is called a data item is expected to be found, perhaps better to error, fail fast and log this, rather than return an empty structure.